component/src/test/java/org/wso2/extension/siddhi/io/file/FileSourceTestCase.java

```diff
@@ -0,0 +1,267 @@
+                "file.uri='" + dir + "/line/logs.txt'," +
+                "file.polling.interval='1000', " +
+                "@map(type='json'))" +
+                "define stream FooStream (symbol string, price float, volume long); " +
+                "define stream BarStream (symbol string, price float, volume long); ";
+
+        String query = "" +
+                "from FooStream " +
+                "select * " +
+                "insert into BarStream; ";
+
+        SiddhiManager siddhiManager = new SiddhiManager();
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(streams + query);
+
+        siddhiAppRuntime.addCallback("BarStream", new StreamCallback() {
+
+            @Override
+            public void receive(Event[] events) {
+                EventPrinter.print(events);
+                int n = count.incrementAndGet();
+
+                for (Event event : events) {
+                    switch (n) {
+                        case 1:
+                            AssertJUnit.assertEquals(0f, event.getData(1));
+                            break;
+                        case 2:
+                            AssertJUnit.assertEquals(1f, event.getData(1));
+                            break;
+                        case 3:
+                            AssertJUnit.assertEquals(2f, event.getData(1));
+                            break;
+                        case 4:
+                            AssertJUnit.assertEquals(3f, event.getData(1));
+                            break;
+                        case 5:
+                            AssertJUnit.assertEquals(4f, event.getData(1));
+                            break;
+                        case 10:
+                            AssertJUnit.assertEquals(9f, event.getData(1));
+                            break;
+                        case 12:
+                            AssertJUnit.assertEquals(11f, event.getData(1));
+                            break;
+                        case 15:
+                            AssertJUnit.assertEquals(14f, event.getData(1));
+                            break;
+                    }
+                }
+            }
+        });
+
+        siddhiAppRuntime.start();
+
+        Thread.sleep(3000);
+
+        //assert event count
+         AssertJUnit.assertEquals("Number of events", 15, count.get());
+        siddhiAppRuntime.shutdown();
+    }
+
+    //from here, tests will be implemented later
+
+    /*@Test
+    public void siddhiIoFileTest5() throws InterruptedException {
+        log.info("test SiddhiIoFile 5");
+        String streams = "" +
+                "@App:name('TestSiddhiApp')" +
+                "@source(type='file',mode='line', tailing='true', " +
+                "file.uri='" + dir + "/tailing/logs.txt'," +
+                "@map(type='json'))" +
+                "define stream FooStream (symbol string, price float, volume long); " +
```

Comment:
By PR #10, 
the commented code segment has been removed and tests have been categorized by the reading mode.